### PR TITLE
Add folder for the Dart Track

### PR DESCRIPTION
**Pull request by Stargator**
Add skeleton structure for the Dart track

### CODE REVIEW

**Comment by Stargator** [comment]
Unfortunately, there's a broken link in the javascript folder. No links are broken in the new dart folder.

File: languages/dart/info/arrow_function_expression.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Arrow Function Expression
+
+An arrow function expression is a syntactically compact alternative to a regular function expression, although without its own bindings to the `this` or `super` keywords.
```
Comment (devkabiir) [inline languages/dart/info/arrow_function_expression.md]: I think this should be mentioned in the transition/from_js.md. Not in the general documentation for arrow functions. The `this` and `super` keywords are only available inside the scope of a class method (including getters/setters/operators/constructors). They are not available to top-level functions.

File: languages/dart/concept-exercises/numbers/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: strings
```
Comment (devkabiir) [inline languages/dart/concept-exercises/numbers/pubspec.yaml]:
```suggestion
name: numbers
```

File: languages/dart/concept-exercises/numbers/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+environment:
+  sdk: '>=2.0.0 <3.0.0'
+dev_dependencies:
+  dart_style: '^1.0.8'
+  test: '>=0.12.24+8 <2.0.0'
```
Comment (devkabiir) [inline languages/dart/concept-exercises/numbers/pubspec.yaml]:
```suggestion
  test: '<2.0.0'
```

File: languages/dart/concept-exercises/futures/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+name: strings
```
Comment (devkabiir) [inline languages/dart/concept-exercises/futures/pubspec.yaml]:
```suggestion
name: futures
```

File: languages/dart/concept-exercises/futures/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+environment:
+  sdk: '>=2.0.0 <3.0.0'
+dev_dependencies:
+  dart_style: '^1.0.8'
+  test: '>=0.12.24+8 <2.0.0'
```
Comment (devkabiir) [inline languages/dart/concept-exercises/futures/pubspec.yaml]:
```suggestion
  test: '<2.0.0'
```

File: languages/dart/keywords/Function.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# `function`
```
Comment (devkabiir) [inline languages/dart/keywords/Function.md]: nit: The filename should be `function.md` to be consistent with other files. 

The `Function` is a built-in type, where as `function` is a keyword. Perhaps we should mention that somewhere.

**Comment by Stargator** [comment]
@devkabiir I am looking to take soem time to address some of this, this week. Thanks for the feedback.

File: languages/dart/README.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
+# Dart
+
+## Primitives
```
Comment (ErikSchierboom) [inline languages/dart/README.md]: For consistency, would it perhaps make sense to have this be 

```suggestion
## Primitive types
```

Comment (ErikSchierboom) [inline languages/dart/README.md]: You could consider creating a h2 heading named `Types`, and then have the individual types be defined as h3 headings. That way, it is easy to see the type sections go together.

File: languages/dart/README.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+[global-object-object]: ./types/object.md
+[global-object-regexp]: ./types/regexp.md
+[global-object-set]: ./types/set.md
+
+## Expressions
+
+- [`new`][keyword-new]
+- `this`
+
+## Object-oriented concepts
```
Comment (ErikSchierboom) [inline languages/dart/README.md]: You could consider creating a h2 heading named `Concepts`, and then have the individual concept categories be defined as h3 headings. That way, it is easy to see the concept sections go together.

File: languages/dart/keywords/Function.md (hunk 6 above)
Comment (Stargator) [reply languages/dart/keywords/Function.md]: Renamed, the keywords can be expanded upon anyone that wants to work on this.

File: languages/dart/types/error.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# `Error`
```
[the remark this answers is not in the document]
Comment (Stargator) [reply languages/dart/types/error.md]: Created exception.md

**Comment by iHiD** [comment]
Whoop whoop! Thanks :)

File: languages/dart/keywords/async.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# `async`
+
+Marks a [function][concept-functions] to be asynchronous function; an asynchronous function is a function which operates asynchronously, using an implicit `Future` to return its result.
```
[the remark this answers is not in the document]
Comment (Stargator) [reply languages/dart/keywords/async.md]: @devkabiir From my reading of the [FutureOf API page](https://api.dart.dev/stable/2.7.0/dart-async/FutureOr-class.html), FutureOf is only a "public stand-in" for a "internal future-or-value generic type". So it's not something we should really use. Instead, Future is the abstract that we should use. So I wouldn't want to confuse folks when they are parts of the same process.

**Pull request merged**